cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,602 @@
+import csv
+import tempfile
+import time
+
+import attr
+from navmazing import NavigateToAttribute
+from navmazing import NavigateToSibling
+from selenium.webdriver.common.keys import Keys
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.widget import Checkbox
+from widgetastic.widget import ConditionalSwitchableView
+from widgetastic.widget import View
+from widgetastic_patternfly import BootstrapSelect
+from widgetastic_patternfly import Button
+from widgetastic_patternfly import PFIcon
+from widgetastic_patternfly import SelectorDropdown
+from widgetastic_patternfly import Text
+from widgetastic_patternfly import TextInput
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.log import logger
+from cfme.utils.version import Version
+from cfme.utils.version import VersionPicker
+from cfme.utils.wait import wait_for
+from widgetastic_manageiq import HiddenFileInput
+from widgetastic_manageiq import MigrationDashboardStatusCard
+from widgetastic_manageiq import MigrationDropdown
+from widgetastic_manageiq import MigrationPlanRequestDetailsList
+from widgetastic_manageiq import MigrationPlansList
+from widgetastic_manageiq import MigrationProgressBar
+from widgetastic_manageiq import RadioGroup
+from widgetastic_manageiq import Table
+from widgetastic_manageiq import V2VPaginatorPane
+
+
+class MigrationView(BaseLoggedInPage):
+
+    @property
+    def in_explorer(self):
+        nav_menu = VersionPicker(
+            {
+                Version.lowest(): ["Compute", "Migration"],
+                "5.10": ["Compute", "Migration", "Migration Plans"],
+            }
+        ).pick()
+        return self.logged_in_as_current_user and self.navigation.currently_selected == nav_menu
+
+    @property
+    def is_displayed(self):
+        return self.in_explorer
+
+
+class DashboardCards(View):
+    not_started_plans = MigrationDashboardStatusCard("Not Started")
+    in_progress_plans = MigrationDashboardStatusCard("In Progress")
+    completed_plans = MigrationDashboardStatusCard("Complete")
+    archived_plans = VersionPicker(
+        {
+            Version.lowest(): MigrationDashboardStatusCard("Infrastructure Mappings"),
+            "5.10": MigrationDashboardStatusCard("Archived"),
+        }
+    )
+
+    def read(self):
+        return [c.read().get("name") for c in self.sub_widgets if c.is_active]
+
+
+class MigrationPlanView(MigrationView):
+    dashboard_cards = View.nested(DashboardCards)
+    plans_list = ConditionalSwitchableView(reference="dashboard_cards")
```

Comment:
There are no views registered against this `ConditionalSwitchableView`, what is the intent here?

Reply:
This is not required @mshriver .
Not implementing conditional switchable here .